src/module.ts

```diff
@@ -0,0 +1,136 @@
+import { defineNuxtModule, useLogger, isNuxt2 } from '@nuxt/kit'
+import { defu } from 'defu'
+import type { SentryCliPluginOptions } from '@sentry/webpack-plugin'
+import { Handlers as SentryHandlers, captureException, withScope } from '@sentry/node'
+import type { ModuleConfiguration } from '../types'
+import type { DeepPartialModuleConfiguration } from '../types/sentry'
+import { envToBool, boolToText, canInitialize, clientSentryEnabled, serverSentryEnabled } from './core/utils'
+import { buildHook, initializeServerSentry, shutdownServerSentry, webpackConfigHook } from './core/hooks'
+
+export interface ModuleOptions extends DeepPartialModuleConfiguration {}
+
+export interface ModulePublicRuntimeConfig extends DeepPartialModuleConfiguration {}
+
+export interface ModulePrivateRuntimeConfig extends DeepPartialModuleConfiguration {}
+
+const logger = useLogger('nuxt:sentry')
+
+export default defineNuxtModule<ModuleConfiguration>({
+  meta: {
+    name: '@nuxtjs/sentry',
+    configKey: 'sentry',
+    compatibility: {
+      nuxt: '^2.15.8',
```

Comment:
I didn't know if this module had any nuxt 2 compatibility constraints so I use the last version, we could probably lower that